tests/rustdoc/bounds.rs

```diff
@@ -18,3 +18,15 @@ pub trait T2 {
     fn f<T: Eq>()
         where Self: Eq, Self: Eq2, T: Eq2;
 }
+
+// Checking that we support empty bounds (we used to crash on empty outlives-bounds).
+// Note that we don't want to hide them since they have a semantic effect.
+// For outlives-bounds, they force the lifetime param to be early-bound instead of late-bound.
+// For trait bounds, it can affect well-formedness (see `ClauseKind::WellFormed`).
+// @has 'foo/fn.empty.html'
+// @has - '//pre[@class="rust item-decl"]' "empty<'a, T>()where T:, 'a:,"
+pub fn empty<'a, T>()
+    where
+        T:,
+        'a:,
+{}
```

Comment:
I gotta run now but I can add more tests demo'ing the importance of showing empty trait bounds, e.g., one involving `generic_const_exprs` & one affecting wf'cking.

Reply:
If you have more regression tests, please add them. The more the better.

Reply:
E.g.,

```rs
#![feature(generic_const_exprs)]

pub fn foo<const N: usize>() where [(); N + 1]: {
    let _: [(); N + 1];
}

fn main() {}
```

cc https://github.com/rust-lang/rust/issues/120994#issuecomment-1939932170

Reply:
And re well-formedness checking (a bit artificial, I know ^^')

```rs
pub fn f() -> <[[[u8]]] as Discard>::Output where <[[[u8]]] as Discard>::Output: {}

pub trait Discard {
    type Output;
}

impl<T: ?Sized> Discard for T {
    type Output = ();
}
```

Reply:
`tests/rustdoc/const-generics/generic_const_exprs.rs` suffices imo

Reply:
Ok!